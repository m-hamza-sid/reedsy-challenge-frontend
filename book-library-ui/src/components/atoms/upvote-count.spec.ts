import { shallowMount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import UpvoteCount from "./upvote-count.vue";
import { config } from "@vue/test-utils";

describe("upvote-count", () => {
  test("should render with upVoteCount prop", () => {
    console.log(config.global.plugins);
    const iUpVoteCount = 2;

    const wrapper = shallowMount(UpvoteCount, { props: { upVoteCount: iUpVoteCount } });

    expect(wrapper.text()).toContain(iUpVoteCount);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should not render anything if count is 0", () => {
    const iUpVoteCount = 0;

    const wrapper = shallowMount(UpvoteCount, { props: { upVoteCount: iUpVoteCount } });

    expect(wrapper.text()).toHaveLength(0);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should not render anything if count is null", () => {
    const iUpVoteCount = null;

    const wrapper = shallowMount(UpvoteCount, { props: { upVoteCount: iUpVoteCount } });

    expect(wrapper.text()).toHaveLength(0);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
