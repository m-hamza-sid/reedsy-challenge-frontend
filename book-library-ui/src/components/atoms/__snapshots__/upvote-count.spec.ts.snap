// Vitest Snapshot v1

exports[`upvote-count > should not render anything if count is 0 1`] = `"<!--v-if-->"`;

exports[`upvote-count > should not render anything if count is null 1`] = `"<!--v-if-->"`;

exports[`upvote-count > should render with upVoteCount prop 1`] = `"<span>atoms.upvote-count.text 2 atoms.upvote-count.times-plural</span>"`;
